.github/workflows/build.yml

```diff
@@ -3,85 +3,80 @@ on:
     branches:
     - main
     - develop
+    - release-*
   pull_request:
     types: [opened, synchronize, reopened]
     branches:
     - main
     - develop
+    - release-*
   workflow_dispatch:
 name: Build and Test Code
 jobs:
   build-artifacts:
-    name: Build and Test Java Artifacts
+    name: Build Java Artifacts
     runs-on: ubuntu-latest
     permissions:
       actions: read
       contents: read
       security-events: write
     steps:
-    - uses: actions/checkout@c85c95e3d7251135ab7dc9ce3241c5835cc595a9 # v3.0.2
+    - uses: actions/checkout@c85c95e3d7251135ab7dc9ce3241c5835cc595a9
       with:
         token: ${{ github.token }}
         submodules: recursive
         fetch-depth: 0
     # -------------------------
-    # Java JDK 11
-    - name: Set up JDK 11
-      uses: actions/setup-java@5ffc13f4174014e2d4d4572b3d74c3fa61aeb2c2 # v3.3.0
+    # Java
+    # -------------------------
+    - name: Set up Maven
+      uses: stCarolas/setup-maven@07fbbe97d97ef44336b7382563d66743297e442f
```

Comment:
Concern: do we need to run a new setup-maven step for caching and that is why we need this new action and run it separately of setup-java?

Reply:
This paves the way for that. It also stabilizes the build environment on a specific version of Maven, which is important for reproducible builds.